storage/innobase/include/dict0mem.h

```diff
@@ -2446,6 +2447,16 @@ struct dict_table_t {
         return true;
     return false;
   }
+
+  /** Iterate through indexes and store their root page
+  in the given list.
+  @param root_pages root page list */
+  void add_index_root_pages(std::unordered_set<uint32_t> *root_pages)
+  {
+    for (auto i= UT_LIST_GET_FIRST(indexes);
+         i != nullptr; i= UT_LIST_GET_NEXT(indexes, i))
+      root_pages->insert(i->page);
+  }
```

Comment:
There is only one user of this function. I think that this can be defined as a `static` non-member function in the compilation unit that needs it. I would not declare it as `inline` because there will be many calls.